src/utils/fileUtils.mjs

```diff
@@ -0,0 +1,39 @@
+import fs from "fs/promises";
+import { BASEDIR } from "../../config.mjs";
+
+export const readFileAsync = async (filePath) => {
+  try {
+    const data = await fs.readFile(filePath, "utf8");
+    return JSON.parse(data);
+  } catch (error) {
+    console.error("Reading file error:", error);
+    throw error;
+  }
+};
+
+export const appendFileAsync = async (filePath, data) => {
+  try {
+    await fs.appendFile(filePath, data + "\n");
+  } catch (error) {
+    console.error("Appending file error:", error);
+    throw error;
+  }
+};
+
+export const getDataFromLogs = async () => {
```

Comment:
This function contain business logic. Your entire file has a folder name `utils`.
So, I can recommend to move business logic to the top level (controller) and this function might be replaces with your own `readFileAsync` function from above
Also I would recommend to rename it as getFileData. No needs to mention is that async, or sync, you function itself is async, so it's understandable already